lib/puppet/provider/x509_cert/openssl.rb

```diff
@@ -10,11 +10,23 @@ def self.private_key(resource)
     file = File.read(resource[:private_key])
     case resource[:authentication]
```

Comment:
For this block I was thinking about refactoring it (prior to your change):
```ruby
cls = case resource[:authentication]
      when :dsa
        OpenSSL::PKey::DSA
      when :rsa
        OpenSSL::PKey::RSA
      when :ec
        OpenSSL::PKey::EC
      else
        raise
      end

cls.new(file, resource[:password])
```

Then with your change it becomes
```ruby
cls = case resource[:authentication]
      when :dsa
        OpenSSL::PKey::DSA
      when :rsa
        OpenSSL::PKey::RSA
      when :ec
        OpenSSL::PKey::EC
      else
        raise
      end

if resource[:password].respond_to?(:unwrap)
  Puppet::Pops::Types::PSensitiveType::Sensitive.new(cls.new(file, resource[:password].unwrap))
else
  cls.new(file, resource[:password])
end
```

But realistically, why don't we use `OpenSSL::Pkey.read(file, resource[:password])` to let OpenSSL do the heavy lifting? Is that method unavailable in older versions? If so, how old?

Reply:
https://github.com/voxpupuli/puppet-openssl/pull/189 does that.

Reply:
I am no Expert with OpenSSL.  I suggest to leave that Change in #189